mono/metadata/locales.c

```diff
@@ -530,22 +532,44 @@ get_current_locale_name (void)
 	return ret;
 }
 
-MonoString*
-ves_icall_System_Globalization_CultureInfo_get_current_locale_name (void)
+static MonoStringHandle
+culture_info_get_current_locale_name (MonoError *error)
 {
-	gchar *locale;
-	MonoString* ret;
+	MonoStringHandle result = MONO_HANDLE_NEW(MonoString, NULL);
+	MONO_HANDLE_ARENA_PUSH ();
+	gchar *locale = NULL;
 	MonoDomain *domain;
 
+	mono_error_init (error);
+
 	locale = get_current_locale_name ();
 	if (locale == NULL)
-		return NULL;
+		goto leave;
 
 	domain = mono_domain_get ();
-	ret = mono_string_new (domain, locale);
+
+	mono_handle_assign (result, mono_string_new_handle (domain, locale, error));
```

Comment:
Yea a handle is basically just a pointer to a `MonoObject*`, but you end up having to allocate a handle in every arena all the way up the call tree just to return.